### DISCUSSION TITLE
this.$patch() method add switch

### DISCUSSION THREAD

**Discussion by ghost** [vuejs/pinia#959] [Help and Questions]
If this.$patch() changes the value of state, it will change the property defined as readonly. How can it be controlled from changing the value of readonly

**Comment by angelhdzmultimedia** [comment]
this.$patch() doesn't change the value of state. It injects the values of the object passed to it as argument in the state's object (hence the "patch" name).
To actually replace the state, one would have to `state.$state = {} ` in JavaScript because JS is dynamic, but in TypeScript one would get an error that {} is not of type "your original state type definition".

![image](https://user-images.githubusercontent.com/10191085/151653235-4639cdca-2c6c-4901-94d2-ac8d792bc6a2.png)
